src/telemetry/storage_usage.rs

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::error::DeviceManagerError;
+use astarte_sdk::types::AstarteType;
+use std::collections::HashMap;
+use sysinfo::{DiskExt, System, SystemExt};
+
+/// get structured data for `io.edgehog.devicemanager.StorageUsage` interface
+pub fn get_storage_usage() -> Result<Vec<HashMap<String, AstarteType>>, DeviceManagerError> {
```

Comment:
You can use a struct instead of an HashMap, since StorageUsage is an object, similar to get_system_status

Reply:
switched to struct and used in getDisks method